tripy/tripy/backend/mlir/executor.py

```diff
@@ -178,8 +179,9 @@ def execute(self, output_devices=List[device], inputs: List["Tensor"] = []) -> L
             out_args.append(memref)
 
         # Execute and populate device pointers.
-        self.session.execute_function("main", in_args=in_args, out_args=out_args, stream=self.stream)
-        self.stream.sync()
+        self.session.execute_function(
+            "main", in_args=in_args, out_args=out_args, stream=Stream.get_active_stream()._stream
```

Comment:
Not sure this would work as expected - if we have a long sequence of dependent operations where they are divided across multiple streams, they will all execute on the stream of only the final operation that is evaluated. Somehow we need to associate the stream with the operation at trace time and potentially construct multiple executables.

Reply:
Usually, multi-stream is well handled within TRT compiler and we shouldn't worry about assigning streams for independent work in the frontend. (unless the user asks for it which means they are aware of what work they want to enqueue on a stream).

Reply:
How does the user ask for it though? If we did:
```py
with tp.Stream() as stream1:
    # do something
    out = ...

with tp.Stream() as stream2:
    out = out + 1
    print(out)
```

Then everything would execute on stream2, even though I expected only the addition on stream2 and everything else on stream1.

Reply:
We can force the evaluation of `out` in `__exit__` of stream1? Though, I was imagining the use of stream only in case of compiled graphs since in the eager mode streams are kind of useless as TRT would not respect the stream for building the engine.

Reply:
Let's merge for now since it works with the limited context of `compiled` execution.